server-common/src/main/java/org/apache/kafka/server/common/StreamsGroupVersion.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import java.util.Map;
+
+public enum StreamsGroupVersion implements FeatureVersion {
+
+    // Version 0 does disable "streams" groups (KIP-1071).
+    SGV_0(0, MetadataVersion.MINIMUM_VERSION, Map.of()),
+
+    // Version 1 enables "streams" groups (KIP-1071).
+    SGV_1(1, MetadataVersion.IBP_4_1_IV2, Map.of());
+
+    public static final String FEATURE_NAME = "streams.group.version";
+
+    // Disabled by default in 4.1 (early access only).
+    public static final StreamsGroupVersion LATEST_PRODUCTION = SGV_0;
```

Comment:
This does not mean "disabled by default". It means that you can not use it unless "unstable features" are allowed.

Reply:
But "unstable features" are disabled by default right :) -- Will update the comment.

Reply:
The aim here is to get to a similar situation for this feature as was achieved for ELR in 4.0. So, yes, right now, you'd need to enable unstable features, but that should not be the end state for 4.1. The update of `MetadataVersion.LATEST_PRODUCTION` is an important future step of this work.